lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3785,6 +3771,124 @@ ParseResult FIRStmtParser::parseInstance() {
   return moduleContext.addSymbolEntry(id, entryId, startTok.getLoc());
 }
 
+/// instance_choice ::=
+///   'inst_choice' id 'of' id id info? newline indent ( id "=>" id )+ dedent
+ParseResult FIRStmtParser::parseInstanceChoice() {
+  auto startTok = consumeToken(FIRToken::kw_instchoice);
+  SMLoc loc = startTok.getLoc();
+
+  // If this was actually the start of a connect or something else handle that.
+  if (auto isExpr = parseExpWithLeadingKeyword(startTok))
+    return *isExpr;
+
+  if (requireFeature(nextFIRVersion, "options"))
+    return failure();
```

Comment:
Same comment as elsewhere about the message here and adding tests.

Also, the usage of `nextFIRVersion` is probably imprecise for this. I think instance choices are going to make 4.0.0?